File: src/main/java/com/bookworm/oopbackend/Payment.java
package com.bookworm.oopbackend;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Data
@Document("payments")
public class Payment {
    @Id
    private String uid;

    private String paymentMethodName;
    private String account;
    private boolean isCredit;
    private LocalDateTime createdAt;

    public Payment (String paymentMethodName, String account, boolean isCredit) {
        this.paymentMethodName = paymentMethodName;
        this.account = account;
        this.isCredit = isCredit;
        this.createdAt = LocalDateTime.now();
    }
}

File: models/Payment.js
const mongoose = require('mongoose');

const paymentSchema = new mongoose.Schema({
  paymentMethodName: {
    type: String,
    required: true,
  },
  account: {
    type: String,
    required: true,
  },
  isCredit: {
    type: Boolean,
    required: true,
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

const Payment = mongoose.model('Payment', paymentSchema);

module.exports = Payment;